lib/datasource/go/index.spec.ts

```diff
@@ -216,19 +254,19 @@ describe('datasource/go', () => {
       httpMock
         .scope('https://some.unknown.website/')
         .get('/example/module?go-get=1')
-        .reply(404);
+        .reply(200);
```

Comment:
there is a test that checks for a 404 returned from the specified URL already, this test should actually be checking that the call was successful but the code doesn't know how to handle the dependency (i.e. not github/gitlab.com/not a gitlab hostRules)